### ISSUE TITLE
shot annotation of RAI dataset not found

### ISSUE THREAD

**Issue by haiahaiah** [soCzech/TransNetV2#45]
Hi, I download RAI from [https://drive.google.com/file/d/1YColUfc3ZuCbiAAHHMYRQVBF2yBikj4N/view](https://drive.google.com/file/d/1YColUfc3ZuCbiAAHHMYRQVBF2yBikj4N/view), but just found videos and scene annotation. I want  to know where do you find the shot boundary annotation. Thanks a lot!

**Comment by AI-lab-sh** [comment]
@soCzech
I have the same issue, any resolve ?

**Comment by Raion-Shin** [comment]
@soCzech I have the same issue. When I list the directory I get the following output:

```
~/TransNetV2# ls RAIDataset/
README.txt    scenes_10.txt  scenes_3.txt  scenes_5.txt  scenes_7.txt  scenes_9.txt
scenes_1.txt  scenes_2.txt   scenes_4.txt  scenes_6.txt  scenes_8.txt  videos
```

**Comment by soCzech** [comment]
You have to transform it into the correct format. Look here https://github.com/soCzech/TransNetV2/blob/master/training/consolidate_datasets.py.

**Comment by Raion-Shin** [comment]
@soCzech Thanks. I changed the path in consolidate_datasets.py to make it work.
```
# RAI_mp4_files = "RAIDataset/*.mp4"
# RAI_txt_files = "RAIDataset/labels/"
RAI_mp4_files = "RAIDataset/videos/*.mp4"
RAI_txt_files = "RAIDataset/"
RAI_target_dir = "consolidated/RAIDataset"
os.makedirs(RAI_target_dir, exist_ok=True)
```

The annotation files seem to be used for scene detection, and the indices start from 1 so I changed the code to subtract the indices by 1.
```
~/TransNetV2# cat npy_datasets/RAIDataset/1.txt 
1 717
718 1590
1591 2333
2334 2488
2489 2733
2734 5576
5577 14541
```
In evaluate.py:
```
gt_scenes = np.loadtxt(np_fn[:-3] + "txt", dtype=np.int32, ndmin=2)
if gt_scenes[0][0] == 1:
    gt_scenes = np.array([[start-1, end-1] for (start, end) in gt_scenes])
```

As a result, I get a pretty low F1 score, which indicates that the processed dataset may be wrong.
(the evaluation code of mine can reproduce the F1 scores on ClipShots and BBCdataset, with the pretrained weight, so I think there's no matter with the code.)
```
    Precision: 9.49%
    Recall:   76.72%
    F1 Score: 16.89%
```
Are they the same files that you used to test your model? I'm afraid the provided link to the RAI dataset has changed over time. Thanks.

**Comment by soCzech** [comment]
Yeah, sorry for that. I have no idea where the issue is. Maybe look at the prediction and the ground truth - maybe there is a slight offset or the files are encoded using different fps.
